File: frontend/demo/component/select/select-overlay-width.ts
import 'Frontend/demo/init'; // hidden-source-line
import '@vaadin/item';
import '@vaadin/list-box';
import '@vaadin/select';
import { html, LitElement } from 'lit';
import { customElement, state } from 'lit/decorators.js';
import type { SelectItem } from '@vaadin/select';
import { getPeople } from 'Frontend/demo/domain/DataService';
import type Person from 'Frontend/generated/com/vaadin/demo/domain/Person';
import { applyTheme } from 'Frontend/demo/theme';

const formatPersonFullName = (person: Person) => `${person.firstName} ${person.lastName}`;

@customElement('select-overlay-width')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    applyTheme(root);
    return root;
  }

  @state()
  private items: SelectItem[] = [];

  protected override async firstUpdated() {
    const { people } = await getPeople({ count: 5 });
    this.items = people.map((person) => ({
      value: formatPersonFullName(person),
      label: `${person.profession} - ${formatPersonFullName(person)}`,
    }));
  }

  protected override render() {
    return html`
      <!-- tag::snippet[] -->
      <vaadin-select
        style="--vaadin-select-overlay-width: 350px"
        label="Employee"
        .items="${this.items}"
      ></vaadin-select>
      <!-- end::snippet[] -->
    `;
  }
}

File: src/main/java/com/vaadin/demo/component/select/SelectOverlayWidth.java
package com.vaadin.demo.component.select;

import java.util.List;

import com.vaadin.demo.DemoExporter; // hidden-source-line
import com.vaadin.demo.domain.DataService;
import com.vaadin.demo.domain.Person;
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.component.select.Select;
import com.vaadin.flow.router.Route;

@Route("select-overlay-width")
public class SelectOverlayWidth extends Div {

    public SelectOverlayWidth() {
        // tag::snippet[]
        Select<Person> select = new Select<>();
        select.setOverlayWidth("350px");
        // end::snippet[]

        List<Person> people = DataService.getPeople(5);
        select.setLabel("Employee");
        select.setItems(people);
        select.setItemLabelGenerator(person -> person.getProfession() + " - "
                + person.getFirstName() + " " + person.getLastName());
        add(select);
    }

    public static class Exporter extends DemoExporter<SelectOverlayWidth> { // hidden-source-line
    } // hidden-source-line
}
